hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.controller.js

```diff
@@ -362,32 +306,202 @@ define([
       });
     }
 
-    function contractChange (reasonId, date) {
-      $scope.$broadcast('hrjc-loader-show');
+    /**
+     * Converts a date string into a Date object (if string is not empty)
+     *
+     * @todo This should probably happen inside the service that returns the data #
```

Comment:
I wonder what does the hash mean?

Reply:
I think it means it needs an instance. The model should return an instance for the contract and the instance should have the dates converted to Date objects instead of strings.